xen/arch/arm/coproc/plat/coproc_xxx.c

```diff
@@ -121,96 +85,41 @@ static const struct mmio_handler_ops vcoproc_xxx_mmio_handler = {
 
 s_time_t vcoproc_xxx_ctx_switch_from(struct vcoproc_instance *curr)
 {
-    s_time_t wait_time;
-
-    if ( !curr )
-        return 0;
-
-    ASSERT(curr->state == VCOPROC_RUNNING ||
-           curr->state == VCOPROC_ASKED_TO_SLEEP);
-
-    wait_time = NOW() & 1 ? coproc_wait_time : 0; /* random for now */
-
-    if ( wait_time == 0 )
-    {
-        if (curr->state == VCOPROC_RUNNING)
-            curr->state = VCOPROC_WAITING;
-        else
-            curr->state = VCOPROC_SLEEPING;
-    }
-
-    return wait_time;
+    /* random for now */
+    return NOW() & 1 ? coproc_wait_time : 0;
 }
 
 static int vcoproc_xxx_ctx_switch_to(struct vcoproc_instance *next)
 {
-    if ( !next )
-        return 0;
-
-    ASSERT(next->state == VCOPROC_WAITING);
-
-    next->state = VCOPROC_RUNNING;
-
+    /* nothing to do */
```

Comment:
if it seems to be a reference code, please put a comment inside, what should be done here